### ISSUE TITLE
README.md is too big. What if it was possible to separate content into multiple .md files?

### ISSUE THREAD

**Issue by AmineAouragh** [ByteByteGoHq/system-design-101#8]
I noticed that the README.md is too big. 

What if it was possible to have all the content structured in separate .md files inside a system-design-101-content parent directory?

Here is an example from the [roadmap.sh](https://github.com/kamranahmedse/developer-roadmap/tree/master/src/data/roadmaps) repository.

That way we could have each topic separately with its own subtopics and own pictures.

A lot easier for contributors to fix anything that need to be fixed or add anything to any topic...

I will think about an easy way to "implement" that and I am open to work on it.

**Comment by tranquanghuy0801** [comment]
Agree with this. I think splitting README into sub-folders would be very helpful to both readers and contributors.

**Comment by MrOnlineCoder** [comment]
I personally disagree, single README is much faster to navigate and easier to read - like a book, I finished one section, and immediately can proceed to next one. Also there is no delay between loading each file separately, at least on Github UI. Or, otherwise, maybe it's possible to perform splitting in the repo, but run a CI/CD or other pipeline that will combine all files into one, so everyone can use whatever method they prefer

**Comment by AmineAouragh** [comment]
@MrOnlineCoder but how about for contributors? it's much easier if you have separate subfolders so you can include changes in only one place and won't have to scroll through a giant README.md file, also commits would be separated depending on which files were updated.

**Comment by MrOnlineCoder** [comment]
@AmineAouragh sure thing, working with such structure is easier, my point was just to have a possibility to still have a full README in the output

**Comment by AmineAouragh** [comment]
you mean even if there was a system-design-101-content parent directory with subfolders you would want to still have the same README.md file?

**Comment by MrOnlineCoder** [comment]
@AmineAouragh not exactly. Many projects have documentation in different formats - full HTML file, HTML file by sections, full PDF, etc. My point is to have something like `README-BUNDLED.md` which can be used by people like me that prefer that way of navigation. The internal or public structure of the files can be whatever

**Comment by AmineAouragh** [comment]
yeah that's what I meant @MrOnlineCoder

**Comment by martin-yin** [comment]
I think dividing the content into multiple md files can allow me to focus more on a certain chapter.
